aea/protocols/dialogue/base.py

```diff
@@ -338,6 +339,52 @@ def __init__(
         )
         self._message_class = message_class
 
+    def __eq__(self, other) -> bool:
+        """Compare two dialogues."""
+        return (
+            type(self) == type(other)  # pylint: disable=unidiomatic-typecheck
+            and self.dialogue_label == other.dialogue_label
+            and self.message_class == other.message_class
+            and self._incoming_messages == other._incoming_messages
+            and self._outgoing_messages == other._outgoing_messages
+            and self.role == other.role
+        )
+
+    def json(self) -> dict:
+        """Get json representation of the dialogue."""
+        data = {
+            "dialogue_label": self._dialogue_label.json,
+            "self_address": self.self_address,
+            "role": self._role.value,
+            "incoming_messages": [i.json() for i in self._incoming_messages],
+            "outgoing_messages": [i.json() for i in self._outgoing_messages],
```

Comment:
what about message class?

Reply:
not possible to serialize this, module and class path can be provided, but it's not as easy just to load it in every case.
in this case, the message class will be restored by the dialogues.